### PR TITLE
RATIS-2106. Add configuration reference for RaftClient

### PR DESCRIPTION
**Pull request by smengcl**
## What changes were proposed in this pull request?

Add config doc for `RaftClient`. Like what RATIS-1732 did for `RaftServer`.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/RATIS-2106

## How was this patch tested?

- n/a

### CODE REVIEW

File: ratis-docs/src/site/markdown/configurations.md [hunk 1]
```diff
@@ -675,4 +675,73 @@ For examples,
...
+|:----------------|:--------------------------------------------------------------|
+| **Description** | minimum number of bytes before data stream flush would happen |
+| **Type**        | SizeInBytes                                                   |
+| **Default**     | 1MB                                                           |
+
+| **Property**    | `raft.client.data-stream.request.timeout` |
+|:----------------|:------------------------------------------|
+| **Description** | timeout for data stream request           |
+| **Type**        | TimeDuration                              |
+| **Default**     | 10000ms                                   |
```
Comment (szetszwo) [inline ratis-docs/src/site/markdown/configurations.md:739]: Could you change it to 10s?
```java
+++ b/ratis-client/src/main/java/org/apache/ratis/client/RaftClientConfigKeys.java
@@ -125,7 +125,7 @@ public interface RaftClientConfigKeys {
     }
 
     String REQUEST_TIMEOUT_KEY = PREFIX + ".request.timeout";
-    TimeDuration REQUEST_TIMEOUT_DEFAULT = TimeDuration.valueOf(10000, TimeUnit.MILLISECONDS);
+    TimeDuration REQUEST_TIMEOUT_DEFAULT = TimeDuration.valueOf(10, TimeUnit.SECONDS);
     static TimeDuration requestTimeout(RaftProperties properties) {
       return getTimeDuration(properties.getTimeDuration(REQUEST_TIMEOUT_DEFAULT.getUnit()),
           REQUEST_TIMEOUT_KEY, REQUEST_TIMEOUT_DEFAULT, getDefaultLog());
```

File: ratis-docs/src/site/markdown/configurations.md [hunk 2]
```diff
@@ -675,4 +675,73 @@ For examples,
...
+| **Type**        | int                                          |
+| **Default**     | 100                                          |
+
+#### Experimental - Configurations related to experimental features.
+
+| **Property**    | `raft.client.async.experimental.send-dummy-request`    |
+|:----------------|:-------------------------------------------------------|
+| **Description** | send a dummy watch request to establish the connection |
+| **Type**        | boolean                                                |
+| **Default**     | true                                                   |
```
Comment (szetszwo) [inline ratis-docs/src/site/markdown/configurations.md:713]: Let's don't add the experimental features.

File: ratis-docs/src/site/markdown/configurations.md [hunk 3]
```diff
@@ -675,4 +675,73 @@ For examples,
...
+
+## Client Configurations
+
+Client configurations are located at `RaftClientConfigKeys`.
+
+### RPC - Configurations related to Client RPC timeout.
+
+| **Property**    | `raft.client.rpc.request.timeout` |
+|:----------------|:----------------------------------|
+| **Description** | timeout for AppendEntries RPC     |
```
Comment (szetszwo) [inline ratis-docs/src/site/markdown/configurations.md:689]: How about `client side timeout for sending a request`?

File: ratis-docs/src/site/markdown/configurations.md [hunk 4]
```diff
@@ -675,4 +675,73 @@ For examples,
...
+| **Type**        | TimeDuration                      |
+| **Default**     | 3000ms                            |
+
+| **Property**    | `raft.client.rpc.watch.request.timeout`      |
+|:----------------|:---------------------------------------------|
+| **Description** | timeout for watch request on the client side |
```
Comment (szetszwo) [inline ratis-docs/src/site/markdown/configurations.md:695]: How about `client side timeout for sending a watch request`?

File: ratis-docs/src/site/markdown/configurations.md [hunk 5]
```diff
@@ -675,4 +675,73 @@ For examples,
...
+
+Client configurations are located at `RaftClientConfigKeys`.
+
+### RPC - Configurations related to Client RPC timeout.
+
+| **Property**    | `raft.client.rpc.request.timeout` |
+|:----------------|:----------------------------------|
+| **Description** | timeout for AppendEntries RPC     |
+| **Type**        | TimeDuration                      |
+| **Default**     | 3000ms                            |
```
Comment (szetszwo) [inline ratis-docs/src/site/markdown/configurations.md:691]: Let's change it to `3s`.
```java
+++ b/ratis-client/src/main/java/org/apache/ratis/client/RaftClientConfigKeys.java
@@ -42,7 +42,7 @@ public interface RaftClientConfigKeys {
     String PREFIX = RaftClientConfigKeys.PREFIX + ".rpc";
 
     String REQUEST_TIMEOUT_KEY = PREFIX + ".request.timeout";
-    TimeDuration REQUEST_TIMEOUT_DEFAULT = TimeDuration.valueOf(3000, TimeUnit.MILLISECONDS);
+    TimeDuration REQUEST_TIMEOUT_DEFAULT = TimeDuration.valueOf(3, TimeUnit.SECONDS);
     static TimeDuration requestTimeout(RaftProperties properties) {
       return getTimeDuration(properties.getTimeDuration(REQUEST_TIMEOUT_DEFAULT.getUnit()),
           REQUEST_TIMEOUT_KEY, REQUEST_TIMEOUT_DEFAULT, getDefaultLog());
```

File: ratis-docs/src/site/markdown/configurations.md [hunk 6]
```diff
@@ -675,4 +675,73 @@ For examples,
...
+
+| **Property**    | `raft.client.rpc.watch.request.timeout`      |
+|:----------------|:---------------------------------------------|
+| **Description** | timeout for watch request on the client side |
+| **Type**        | TimeDuration                                 |
+| **Default**     | 10000ms                                      |
```
Comment (szetszwo) [inline ratis-docs/src/site/markdown/configurations.md:697]: Let's change it to `10s`.
```java
+++ b/ratis-client/src/main/java/org/apache/ratis/client/RaftClientConfigKeys.java
@@ -53,7 +53,7 @@ public interface RaftClientConfigKeys {
 
     String WATCH_REQUEST_TIMEOUT_KEY = PREFIX + ".watch.request.timeout";
     TimeDuration WATCH_REQUEST_TIMEOUT_DEFAULT =
-        TimeDuration.valueOf(10000, TimeUnit.MILLISECONDS);
+        TimeDuration.valueOf(10, TimeUnit.SECONDS);
     static TimeDuration watchRequestTimeout(RaftProperties properties) {
       return getTimeDuration(properties.getTimeDuration(WATCH_REQUEST_TIMEOUT_DEFAULT.getUnit()),
           WATCH_REQUEST_TIMEOUT_KEY, WATCH_REQUEST_TIMEOUT_DEFAULT, getDefaultLog());
```

**Review by szetszwo — COMMENTED** [review]
@smengcl , thanks for working on this!  Please see the comments inlined.

File: ratis-docs/src/site/markdown/configurations.md (hunk 1 above)
Comment (smengcl) [reply ratis-docs/src/site/markdown/configurations.md:739]: Sure!

File: ratis-docs/src/site/markdown/configurations.md (hunk 3 above)
Comment (smengcl) [reply ratis-docs/src/site/markdown/configurations.md:689]: Good idea. Done

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by smengcl** [comment]
Thanks @szetszwo for reviewing and merging this

**Pull request merged**